src/lib/core.d.ts

```diff
@@ -836,7 +836,7 @@ interface RegExp {
     lastIndex: number;
 
     // Non-standard extensions
-    compile(): RegExp;
+    compile(): this;

```

Comment:
According to ES specs it should return the `this` value of the `compile()` call. 
So using `this` type here seems like the correct thing to do.

BTW: this method is deprecated so hopefully it won't matter much.